sections/main-product.liquid

```diff
@@ -48,262 +52,42 @@
-                        {% if featured_media.preview_image.width >= 54 %}{{ featured_media.preview_image | image_url: width: 54 }} 54w,{% endif %}
-                        {% if featured_media.preview_image.width >= 74 %}{{ featured_media.preview_image | image_url: width: 74 }} 74w,{% endif %}
-                        {% if featured_media.preview_image.width >= 104 %}{{ featured_media.preview_image | image_url: width: 104 }} 104w,{% endif %}
-                        {% if featured_media.preview_image.width >= 162 %}{{ featured_media.preview_image | image_url: width: 162 }} 162w,{% endif %}
-                        {% if featured_media.preview_image.width >= 208 %}{{ featured_media.preview_image | image_url: width: 208 }} 208w,{% endif %}
-                        {% if featured_media.preview_image.width >= 324 %}{{ featured_media.preview_image | image_url: width: 324 }} 324w,{% endif %}
-                        {% if featured_media.preview_image.width >= 416 %}{{ featured_media.preview_image | image_url: width: 416 }} 416w,{% endif %},
-                        {{ featured_media.preview_image | image_url }} {{ media.preview_image.width }}w
-                      "
-                      src="{{ featured_media | image_url: width: 416 }}"
-                      sizes="(min-width: {{ settings.page_width }}px) calc(({{ settings.page_width | minus: 100 | times: media_width | round }} - 4rem) / 4), (min-width: 990px) calc(({{ media_width | times: 100 }}vw - 4rem) / 4), (min-width: 750px) calc((100vw - 15rem) / 8), calc((100vw - 14rem) / 3)"
-                      alt="{{ featured_media.alt | escape }}"
-                      height="208"
-                      width="208"
-                      loading="lazy"
-                    >
-                  </button>
+          <div class="glide__bullets" data-glide-el="controls[nav]">
+            {% for image in product.images %}
+              <button class="glide__bullet" data-glide-dir="={{ image.position | minus: 1 }}"></button>
+            {% endfor %}
+          </div>
+          <div class="glide__arrows" data-glide-el="controls">
+             <button class="glide__arrow glide__arrow--left product-images-arrow--left" data-glide-dir="<">prev</button>
+             <button class="glide__arrow glide__arrow--right product-images-arrow--right" data-glide-dir=">">next</button>
+          </div>
+        </div>
+        <div class="glide glide--thumbnails" data-glide-thumbnails>
+          <div class="glide__track glide__track--thumbnails" data-glide-el="track">
+            <ul class="glide__slides glide__slides--thumbnails" data-glide-thumbnail-slides="{{ product.images | size }}">
+              {% for image in product.images %}
+                <li class="glide__slide glide__slide--thumbnails" data-thumbnail-index="{{ forloop.index0 }}">
+                  <img src="{{ image | image_url: width: "120x"}}" loading="lazy" alt="{{ image.alt }}" class="image--thumbnails" data-thumbnail-image>
```

Comment:
Add native lazyloading to any image that isn't the biggest image a user will see when they first load the page.